wordpress-mobile/WordPress-Android
# Tooltip onboarding announcement for IA create FAB

Fixes #11012

This PR adds an announcing tooltip for the create FAB in the My Site page.

The logic is the following:

- When this new app version is installed (or updated from previous older installation) the tooltip is shown near to the FAB

<p align="center">
  <img width="300" src=https://user-images.githubusercontent.com/47797566/72091217-f5d31980-330f-11ea-8c89-a5bb4082a231.png>
</p>

- The tooltip is presented near the FAB until the user first tap on the FAB itself
- Once tapped on the FAB the tooltip is not shown any more and there is no way to have it back unless the app storage data is cleared or the user fully uninstall and reinstall the app

## Notes

- The tooltip work with RTL languages

<p align="center">
  <img width="300" src=https://user-images.githubusercontent.com/47797566/72091155-d3410080-330f-11ea-812c-463b277483c1.png>
</p>

- Current implementation of `WPTooltipView` is pretty basic. Mainly to cover this specific scenario. More work/rework is needed to make it a full custom view tooltip component. I explored a bit also a more dynamic approach with the PopupWindow but messing around only with simple layouts seemed more convenient in this simple scenario. Also the placement and interactions (like the coordinator layout behavior with snackbars like in this case) comes at not cost in this way. (Nonetheless the PopupWindow approach can be something to further explore later and I have something basic working also there) 
- I used Chris original position and copy of the tooltip but we can change it (ping @osullivanchris)

## To test
Again all the feature is still behind feature flag so:

- First smock test with vanilla to check that no evident issues were introduced in the My Site and switching between pages in the `WPMainActivity`
- Compile zalpha flavor and check that you get the tooltip near the FAB
- Use the app without using the FAB (for example create a post from Blog Posts) and check that the tooltip is always presented when you come back to the My Site page
- Use the FAB and check that after that the tooltip is no more visualized. Also close and open the app to confirm.

PR submission checklist:
